content/contribute/contributing-and-managing-patterns.adoc

```diff
@@ -0,0 +1,145 @@
+---
+menu: contribute
+title: Contributing and managing patterns
+weight: 50
+aliases: /contributing-and-managing-patterns/
+---
+
+:toc:
+:_content-type: ASSEMBLY
+include::modules/comm-attributes.adoc[]
+
+[id="contributing to patterns"]
+= Contributing to patterns
+
+To contribute, add the upstream repository as an additional remote and work on a branch other than main. This approach allows you to merge changes into the main branch (reflected in GitOps workflows) and makes upstream contributions easier. Contributions from your forked main branch typically include:
+
+. Customizations to `values-global.yaml` and other installation-specific files.
+. Commits made by Tekton and other automated processes unique to your setup.
+
+To isolate changes for upstream contributions (`hcp` is "Validated Patterns"), you can use a separate remote and branch name:
+
+[source,terminal]
+----
+$ git remote add hcp https://github.com/validatedpatterns/industrial-edge
+$ git fetch --all
+$ git branch -b hcp-main -t hcp/main
+   <make changes on the hcp-main branch>
+$ git push origin hcp-main
+----
+
+To update `hcp-main` branch with upstream changes:
+
+[source,terminal]
+----
+$ git checkout hcp-main
+$ git pull --rebase
+----
+
+To reflect these changes in your forked repository (useful for submitting a PR later):
+
+[source,terminal]
+----
+$ git push origin hcp-main
+----
+
+To integrate upstream pattern changes into your local GitOps process:
+
+[source,terminal]
+----
+$ git checkout main
+$ git merge hcp-main
+$ git push origin main
+----
+
+This workflow ensures that the `hcp-main` branch:
+
+. Remains separate from changes made by your local GitOps processes.
+. Can be merged or cherry-picked into your local main branch for use in GitOps workflows (useful for tracking submodule updates, such as common).
+. Serves as a solid basis for Pull Requests upstream, as it will not include local configuration differences or local GitOps commits.
+
+[id="changing-subtrees"]
+== Changing subtrees
+
+Our patterns use the Git subtree feature to promote modularity, enabling multiple patterns to share the same common base. Over time, we move more functionality into the common directory to isolate pattern-specific components as standard usage practices emerge. This approach strengthens the tools in the common directory, adds features, and simplifies the development of new patterns. Typically, we maintain the common subtree during routine updates, and pulling changes from upstream includes any updates from the common directory.
```

Comment:
NOt crazy on the "we" stuff here can we rewrite with "The Git subtree feature promotes modularity by enabling multiple patterns to share a common base. Over time, functionality is moved into the common directory to isolate pattern-specific components as standard usage practices emerge. This approach strengthens tools in the common directory, adds features, and simplifies the development of new patterns. The common subtree is typically maintained during routine updates, and pulling changes from upstream includes any updates from the common directory."